tests/test_helper.go

```diff
@@ -0,0 +1,212 @@
+			Name:      TestObjectName,
+			Namespace: TestNamespace,
+		},
+		Spec: migapi.MigPlanSpec{
+			Namespaces: namespaces,
+			SrcMigClusterRef: &core.ObjectReference{
+				Namespace: TestNamespace,
+				Name:      TestObjectName,
+			},
+			DestMigClusterRef: &core.ObjectReference{
+				Namespace: TestNamespace,
+				Name:      TestDestinationCluster,
+			},
+			MigStorageRef: &core.ObjectReference{
+				Name:      TestObjectName,
+				Namespace: TestNamespace,
+			},
+		},
+	}
+}
+
+func NewMigStorage() (*migapi.MigStorage, *core.Secret) {
+	return &migapi.MigStorage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TestObjectName,
+				Namespace: TestNamespace,
+			},
+			Spec: migapi.MigStorageSpec{
+				BackupStorageConfig: migapi.BackupStorageConfig{
+					CredsSecretRef: &core.ObjectReference{
+						Namespace: ConfigNamespace,
+						Name:      TestStorageSecret,
+					},
+					AwsBucketName: os.Getenv(AWSBUCKETNAME),
+				},
+			BackupStorageProvider: os.Getenv(BACKUPSTORAGEPROVIDER),
+				// TODO: define system variable. can these both be different?
+				VolumeSnapshotProvider: os.Getenv(BACKUPSTORAGEPROVIDER),
+				VolumeSnapshotConfig: migapi.VolumeSnapshotConfig{
+					CredsSecretRef: &core.ObjectReference{
+						Namespace: ConfigNamespace,
+						Name:      TestStorageSecret,
+					},
+				},
+			},
+		}, &core.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TestStorageSecret,
+				Namespace: ConfigNamespace,
+			},
+			Type: TestSecretType,
+			Data: map[string][]byte{
+				"aws-access-key-id":     []byte(os.Getenv(AWSACCESSKEY)),
+				"aws-secret-access-key": []byte(os.Getenv(AWSSECRETKEY)),
+			},
+		}
+}
+
+func NewMigCluster(saToken []byte) (*migapi.MigCluster, *core.Secret) {
+	return &migapi.MigCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TestObjectName,
+				Namespace: TestNamespace,
+			},
+			Spec: migapi.MigClusterSpec{
+				IsHostCluster: false,
+				URL:           os.Getenv(SOURCEURL),
+				ServiceAccountSecretRef: &core.ObjectReference{
+					Namespace: ConfigNamespace,
+					Name:      TestClusterSecret,
+				},
+				Insecure:            true,
+				ExposedRegistryPath: os.Getenv(EXPOSEDREGISTRYPATH),
+			},
+		}, &core.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TestClusterSecret,
+				Namespace: ConfigNamespace,
+			},
+			Data: map[string][]byte{
+				// TODO: Get the token with kubecofig of source
+				"saToken": saToken,
+			},
+			Type: TestSecretType,
+		}
+}
+
+func NewMigrationController() *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "migration.openshift.io/v1alpha1",
+			"kind":       "MigrationController",
+			"metadata": map[string]interface{}{
+				"name":      MigrationController,
+				"namespace": TestNamespace,
+			},
+			// TODO: take variables from env
+			"spec": map[string]interface{}{
+				"velero_plugin_fqin":        "quay.io/konveyor/openshift-velero-plugin:latest",
```

Comment:
you'd need a way to override this?

Reply:
Do you mean take the values of variables from user/env right?

Reply:
yes